strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/layout/npm/NpmArtifactController.java

```diff
@@ -346,6 +355,14 @@ public void downloadPackage(@RepositoryMapping Repository repository,
         final String storageId = repository.getStorage().getId();
         final String repositoryId = repository.getId();
 
+        if (!packageNameWithVersion.startsWith(packageName + "-"))
+        {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        }
+
+        //Example of packageNameWithVersion core-9.0.1-next.8.tgz
+        final String packageVersion = packageNameWithVersion.substring(packageName.length() + 1, packageNameWithVersion.length());
```

Comment:
we can have a helper method for this and use it within `downloadPackageWithScope` as well